activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -250,8 +258,12 @@ def recreate_database(name, options = {})
       def create_database(name, options = {})
         if options[:collation]
           execute "CREATE DATABASE #{quote_table_name(name)} DEFAULT COLLATE #{quote_table_name(options[:collation])}"
+        elsif options[:charset]
+          execute "CREATE DATABASE #{quote_table_name(name)} DEFAULT CHARACTER SET #{quote_table_name(options[:charset])}"
+        elsif supports_longer_index_key_prefix?
+          execute "CREATE DATABASE #{quote_table_name(name)} DEFAULT CHARACTER SET `utf8mb4`"
         else
-          execute "CREATE DATABASE #{quote_table_name(name)} DEFAULT CHARACTER SET #{quote_table_name(options[:charset] || 'utf8mb4')}"
+          raise "Configure a supported :charset in config/database.yml and ensure innodb_large_prefix is enabled to support indexes on varchar(255) string columns."
```

Comment:
Should we mention about `config/database.yml` in this method?
`create_database` could work without `config/database.yml`.

Reply:
I see. Removed `config/database.yml` from the message.